Python/thread_nt.h

```diff
@@ -266,6 +266,15 @@ PyThread_exit_thread(void)
     _endthreadex(0);
 }
 
+void _Py_NO_RETURN
+_PyThread_hang_thread(void)
+{
+    dprintf(("%lu: PyThread_hang_thread called\n", PyThread_get_thread_ident()));
+    while (1) {
+        Sleep(INFINITE);
+    }
```

Comment:
Nothing can interrupt `Sleep()`, so the loop isn't needed. 

Alternatively, suspend the thread via `SuspendThread(GetCurrentThread())`. The latter increments the thread suspend count by one, and `ResumeThread(hThread)` decrements it by one. The thread doesn't execute again until the suspend count is decremented to 0.

Or call `SleepEx(INFINITE, TRUE)` to wait in an alertable state. In this state, the thread will wake up to execute a queued asynchronous procedure call (APC) from `QueueUserAPC()`, or from I/O completion due to a previous `ReadFileEx()` or `WriteFileEx()` call. A loop is necessary in this case because `SleepEx()` returns with `STATUS_USER_APC` (i.e. `WAIT_IO_COMPLETION`) when an APC is executed.

Reply:
Done.